tpm2tools/flush_active_handles.go

```diff
@@ -0,0 +1,40 @@
+package tpm2tools
+
+
+import (
+    "io"
+    "github.com/google/go-tpm/tpm2"
+    "github.com/google/go-tpm/tpmutil"
+)
+
+// FlushActiveHandles calls FlushContext() on all active handles within the 
+// TPM at io.ReadWriter rw. Returns nil if successful or TPM has no active
+// handles.
+func FlushActiveHandles(rw io.ReadWriter) error {
+    handles, err := activeHandles(rw)
+    if err != nil {
+        return err
+    }
+    for _, handle := range handles {
+        err = tpm2.FlushContext(rw, handle)
+        if err != nil {
+            return err
+        }
+    }
+    return nil
+}
+
+// activeHandles returns a slice of tpmutil.Handle objects of all handles
+// active within the TPM at ioReadWriter rw.
+func activeHandles(rw io.ReadWriter) ([]tpmutil.Handle, error) {
```

Comment:
This is the main function which is useful at the library level. It would be nice to have a way to get a slice of all the handles with some specific property. However, this function isn't correct as written.